Update contribution guidelines with AI usage rules (#440)
holoviz/holoviz: doc/contribute.md

```

## Using AI README

**Maintainers are humans and have finite time and energy to review contributions**. As such, they need to prioritize their work, and they may not have the time to review all contributions, especially if they are of low effort. We especially value contributions that arise from real user use cases, i.e. actually using HoloViz packages like Lumen, Panel, HoloViews, hvPlot, Datashader, and others to build something you care about, but encounter a bug or missing feature.

As an example, Andrew saw a cool chart online and tried to reproduce it with HoloViews, but noticed that the labels' text color doesn't cycle through alongside the points' color. This motivated him to [create an issue to report it](https://github.com/holoviz/holoviews/issues/5887) and separately [create a PR to fix it](https://github.com/holoviz/holoviews/pull/5888).

Remember, contributions don't just have to be Pull Request(s) (PR)s either! Filing bugs, helping others on Discord or Discourse, writing blog posts, or sharing things you built on social media all count! See below more ideas!

> If you are a new contributor and are contributing a PR though, you can use AI tools (in fact HoloViz has its own [MCP](https://marcskovmadsen.github.io/holoviz-mcp/), **but they must follow these rules in the PR description**:

Author: ```suggestion
If you are a new contributor and are contributing a PR though, you can use AI tools (in fact HoloViz has its own [MCP](https://marcskovmadsen.github.io/holoviz-mcp/)), **but they must follow these rules in the PR description**:
```

> If you are a new contributor and are contributing a PR though, you can use AI tools (in fact HoloViz has its own [MCP](https://marcskovmadsen.github.io/holoviz-mcp/)), **but they must follow these rules in the PR description**:

Author: ```suggestion
If you are a new contributor and are contributing a PR though, you can use AI tools (in fact HoloViz has its own [MCP](https://marcskovmadsen.github.io/holoviz-mcp/)), **but you must follow these rules in the PR description**:
```

- Follow the repos' PR template
- Explain what motivated the PR (were you trying to achieve something, but encountered ...?)
- Do NOT list out what changed, unless you want to bring attention to specific changes
- Include reproduction steps, ideally an Minimal Reproducible Verifiable Example
- Add screenshots or screen recordings (before changes and after changes) must be included
- Cite sources and docs you referenced with working links

> Please limit to 2 open PRs at a time across our repos; if you have more than 2 open PRs. Maintainers reserve the right to close PRs they deem low-effort or AI-generated without review.

Author: ```suggestion
Please limit to 2 open PRs at a time across our repos; if you have more than 2 open PRs.

Separately, please refrain from tagging maintainers shortly after opening a PR/issue, as reviews will happen when time allows. Only bump a PR/issue that has been open for a longer period of time (weeks) without a response.

Maintainers reserve the right to close PRs they deem low-effort or AI-generated without review.
```

Contributing to the HoloViz project might even get you a job, which has been the case for a couple of HoloViz members.
```

> ## Using AI README
> 

Author: ```suggestion
```

Contributing to the HoloViz project might even get you a job, which has been the case for a couple of HoloViz members.
```

## Using AI README

> **Maintainers are humans and have finite time and energy to review contributions**. As such, they need to prioritize their work, and they may not have the time to review all contributions, especially if they are of low effort. We especially value contributions that arise from real user use cases, i.e. actually using HoloViz packages like Lumen, Panel, HoloViews, hvPlot, Datashader, and others to build something you care about, but encounter a bug or missing feature.

Reviewer: ```suggestion
**Maintainers are humans and have finite time and energy to review contributions**. As such, they need to prioritize their work, and they may not have the time to review all contributions. We will be especially reluctant to take time to review submissions that seem to be of low effort on the part of the submitter. Conversely, we especially value contributions that arise from real user use cases, i.e. actually using HoloViz packages like Lumen, Panel, HoloViews, hvPlot, Datashader, and others to build something you care about, but encounter a bug or missing feature.
```

As an example, Andrew saw a cool chart online and tried to reproduce it with HoloViews, but noticed that the labels' text color doesn't cycle through alongside the points' color. This motivated him to [create an issue to report it](https://github.com/holoviz/holoviews/issues/5887) and separately [create a PR to fix it](https://github.com/holoviz/holoviews/pull/5888).

> Remember, contributions don't just have to be Pull Request(s) (PR)s either! Filing bugs, helping others on Discord or Discourse, writing blog posts, or sharing things you built on social media all count! See below more ideas!

Author: ```suggestion
Remember, contributions don't just have to be Pull Request(s) (PR)s either! Filing bugs, helping others on Discord or Discourse, writing blog posts, or sharing things you built on social media all count! See below for more ideas!
```

> Remember, contributions don't just have to be Pull Request(s) (PR)s either! Filing bugs, helping others on Discord or Discourse, writing blog posts, or sharing things you built on social media all count! See below for more ideas!

Reviewer 2: I think these two sections should be moved a bit down.

```
Maintainers reserve the right to close PRs they deem low-effort or AI-generated without review.

Remember, contributions don't just have to be Pull Request(s) (PR)s either! Filing bugs, helping others on Discord or Discourse, writing blog posts, or sharing things you built on social media all count! See below for more ideas!

As an example, Andrew saw a cool chart online and tried to reproduce it with HoloViews, but noticed that the labels' text color doesn't cycle through alongside the points' color. This motivated him to [create an issue to report it](https://github.com/holoviz/holoviews/issues/5887) and separately [create a PR to fix it](https://github.com/holoviz/holoviews/pull/5888).

Let's continue building a meaningful community!
```

> If you are a new contributor and are contributing a PR though, you can use AI tools (in fact HoloViz has its own [MCP](https://marcskovmadsen.github.io/holoviz-mcp/)), **but you must follow these rules in the PR description**:

Reviewer 2: I'm not sure we should mention the MCP. Please correct me if I'm wrong, but the MCP is about how to use HoloViz, and not on how to contribute to it, and therefore could cause confusion.

- Follow the repos' PR template
- Explain what motivated the PR (were you trying to achieve something, but encountered ...?)
- Do NOT list out what changed, unless you want to bring attention to specific changes

> - Include reproduction steps, ideally an Minimal Reproducible Verifiable Example

Reviewer: ```suggestion
- Include reproduction steps, ideally a Minimal Reproducible Verifiable Example
```

> - Add screenshots or screen recordings (before changes and after changes) must be included

Reviewer: ```suggestion
- Screenshots or screen recordings (before changes and after changes) must be included
```

- Cite sources and docs you referenced with working links

> Please limit to 2 open PRs at a time across our repos; if you have more than 2 open PRs.

Reviewer: >

Author: Seems like you left an empty comment!
<img width="821" height="428" alt="image" src="https://github.com/user-attachments/assets/aeb2d8e7-b1b0-446b-bb09-66963020f4a5" />

Author: Oh I see now! It was an incomplete thought. Fixed it.

> - Cite sources and docs you referenced with working links

Reviewer 2: Maybe a bullet point about not copy/paste the full output of AI to issues and PR, as this is not helping (IMO). We want a curated description of what have happened by the user itself.

> Please limit to 2 open PRs at a time across our repos; if you have more than 2 open PRs.

Author: ```suggestion
Please limit to 2 open PRs at a time across our repos; if you have more than 2 open PRs, they may be closed, especially if it is your first time contributing to HoloViz.
```
